client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -878,7 +881,17 @@ function mediaButton( editor ) {
 	editor.on( 'touchstart touchmove touchend', selectImageOnTap() );
 
 	editor.on( 'init', function () {
-		MediaStore.on( 'change', updateMedia );
+		unsubscribeFromReduxStore = store.subscribe( () => {
+			const { ID: siteId } = getSelectedSite( store.getState() );
+			if ( ! siteId ) {
+				return;
+			}
```

Comment:
I'd use `getSelectedSiteId` there.

If the return value of `getSelectedSite` is `null`, which is probably impossible to achieve (active editor without a selected site and post?), the `{ ID }` destructuring will throw. It's more correct to check the object and only then access its `ID` property.